src/lib/auth.ts

```diff
@@ -22,12 +23,22 @@ export interface session extends Session {
     role: string;
     email: string;
     name: string;
+    provider: string;
+    g_username?: string;
```

Comment:
Whenever you want to use g_username. you can query to db from server side. Adding all info in session does not make sense.

Reply:
The problem here is, connecting github to the existing users' credentials. So session is required to identify them and it's only accessible during github login and not again ever. So maybe it's required.